### PR TITLE
Added useful procs to strutils for string testing.

### PR DESCRIPTION
**Pull request by jyapayne**
Added procs:

    isAlpha
    isAlphaNumeric
    isDigit
    isSpace
    isUpper
    isLower

### CODE REVIEW

File: lib/pure/strutils.nim [hunk 1]
```diff
@@ -60,6 +60,132 @@ const
...
+  ##
+  ## This checks a-z, A-Z ASCII characters only.
+  return c in Letters
+
+proc isAlphaNumeric*(c: char): bool {.noSideEffect, procvar,
+  rtl, extern: "nsuIsAlphaNumericChar".}=
+  ## Checks whether or not `c` is alphanumeric.
+  ##
+  ## This checks a-z, A-Z, 0-9 ASCII characters only.
+  return c in Letters and c in Digits
```
Comment (dom96) [inline lib/pure/strutils.nim]: Should this be an ``or``?

**Comment by dom96** [comment]
Please add tests for these procs to the bottom of strutils.

**Comment by jyapayne** [comment]
Added tests and put changed the "and" to an "or" as per your advice :)

File: lib/pure/strutils.nim [hunk 2]
```diff
@@ -1526,3 +1652,55 @@ when isMainModule:
   doAssert strip("sfoofoofoos", trailing = false, chars = {'s'}) == "foofoofoos"
 
   doAssert "  foo\n  bar".indent(4, "Q") == "QQQQ  foo\nQQQQ  bar"
+
+  doAssert isAlpha('r') == true
```
Comment (dom96) [inline lib/pure/strutils.nim]: The ``== true`` is redundant ;)

File: lib/pure/strutils.nim [hunk 3]
```diff
@@ -1526,3 +1652,55 @@ when isMainModule:
...
+  doAssert isAlpha('A') == true
+  doAssert isAlpha('$') == false
```
Comment (dom96) [inline lib/pure/strutils.nim]: ``doAssert(not isAlpha('$'))``

File: lib/pure/strutils.nim (hunk 2 above)
Comment (jyapayne) [inline lib/pure/strutils.nim]: You are correct :)

File: lib/pure/strutils.nim (hunk 3 above)
Comment (jyapayne) [inline lib/pure/strutils.nim]: That looks much better. Thanks for the pointers :)

**Comment by Araq** [comment]
Argh, can we revert this please? I don't like them at all. All we need is 'allCharsOf' which then takes a set. And I think something like this already exists. No further bloat for strutils please.

**Comment by kirbyfan64** [comment]
Why? Many other programming languages have things like this, including Python (`str.isalpha` and `str.isnumeric`). Besides, this could later be extended to support Unicode.

**Comment by jyapayne** [comment]
I think they are useful as well. Before I added them, I did a grep to see if they existed in the std lib. I did not find any in the std lib, but I did find them somewhere in the compiler source code.

**Comment by Araq** [comment]
Many other programming languages do need it because they lack 'set[char]'. Nim has 'set[char]' and so doesn't need it. These useful helpers are also not Unicode aware, so they don't give us anything over the 'set[char]' based solution. Nim is not Python, it is better. ;-)

**Pull request merged**